servlet/src/main/scala/org/http4s/servlet/Http4sServlet.scala

```diff
@@ -83,7 +81,7 @@ class Http4sServlet(service: HttpService,
                             request: Request,
                             bodyWriter: BodyWriter): Task[Unit] = {
     ctx.addListener(new AsyncTimeoutHandler(request, bodyWriter))
-    val response = Task.fork(serviceFn(request))(threadPool)
+    val response: Task[Response] = Task.start(serviceFn(request))(Strategy.fromExecutor(threadPool)).unsafeRun
```

Comment:
This `unsafeRun` in the middle looks suspicious.  We have a `Task[Task[Response]]` now, right?  Can we just `.join` or `.flatten` or `.flatMap(identity)` or whatever compiles here?